frame/support/procedural/src/default_no_bound.rs

```diff
@@ -25,72 +26,122 @@ pub fn derive_default_no_bound(input: proc_macro::TokenStream) -> proc_macro::To
+
 	let (impl_generics, ty_generics, where_clause) = input.generics.split_for_impl();
 
 	let impl_ = match input.data {
-		syn::Data::Struct(struct_) => match struct_.fields {
-			syn::Fields::Named(named) => {
-				let fields = named.named.iter().map(|i| &i.ident).map(|i| {
-					quote::quote_spanned!(i.span() =>
-						#i: core::default::Default::default()
-					)
+		Data::Struct(struct_) => match struct_.fields {
+			Fields::Named(named) => {
+				let fields = named.named.iter().map(|field| &field.ident).map(|ident| {
+					quote_spanned! {ident.span() =>
+						#ident: core::default::Default::default()
+					}
 				});
 
-				quote::quote!( Self { #( #fields, )* } )
+				quote! { Self { #( #fields, )* } }
 			},
-			syn::Fields::Unnamed(unnamed) => {
-				let fields =
-					unnamed.unnamed.iter().enumerate().map(|(i, _)| syn::Index::from(i)).map(|i| {
-						quote::quote_spanned!(i.span() =>
-							core::default::Default::default()
-						)
-					});
+			Fields::Unnamed(unnamed) => {
+				let fields = unnamed.unnamed.iter().map(|field| {
+					quote_spanned! {field.span()=>
+						core::default::Default::default()
+					}
+				});
 
-				quote::quote!( Self ( #( #fields, )* ) )
+				quote! { Self ( #( #fields, )* ) }
 			},
-			syn::Fields::Unit => {
-				quote::quote!(Self)
+			Fields::Unit => {
+				quote! { Self }
 			},
 		},
-		syn::Data::Enum(enum_) =>
-			if let Some(first_variant) = enum_.variants.first() {
-				let variant_ident = &first_variant.ident;
-				match &first_variant.fields {
-					syn::Fields::Named(named) => {
-						let fields = named.named.iter().map(|i| &i.ident).map(|i| {
-							quote::quote_spanned!(i.span() =>
-								#i: core::default::Default::default()
-							)
-						});
-
-						quote::quote!( #name :: #ty_generics :: #variant_ident { #( #fields, )* } )
-					},
-					syn::Fields::Unnamed(unnamed) => {
-						let fields = unnamed
-							.unnamed
-							.iter()
-							.enumerate()
-							.map(|(i, _)| syn::Index::from(i))
-							.map(|i| {
-								quote::quote_spanned!(i.span() =>
+		Data::Enum(enum_) => {
+			if enum_.variants.is_empty() {
+				return quote! { Self }.into();
+			}
+
+			let tagged_variants = enum_
+				.variants
+				.into_iter()
+				.flat_map(|v| {
+					v.attrs
+						.clone()
+						.into_iter()
+						.filter(|a| a.path.is_ident("default"))
+						.map(move |_| v.clone())
+				})
+				.collect::<Vec<_>>();
+
+			match &*tagged_variants {
+				[] => {
+					return syn::Error::new(name.clone().span(), "no #[default] attribute found")
+						.into_compile_error()
+						.into()
+				},
+				[default_variant] => {
+					let variant_ident = &default_variant.ident;
+
+					let fully_qualified_variant_path =
+						quote! { <#name #ty_generics>::#variant_ident };
+
+					match &default_variant.fields {
+						Fields::Named(named) => {
+							let fields =
+								named.named.iter().map(|field| &field.ident).map(|ident| {
+									quote_spanned! {ident.span()=>
```

Comment:
```suggestion
									quote_spanned! { ident.span()=>
```

Reply:
This is the intended syntax of `quote_spanned!`: https://docs.rs/quote/latest/quote/macro.quote_spanned.html#syntax